tests/test_providers.py

```diff
@@ -552,6 +553,27 @@ def test_iacr_3(self):
         filename = prov.run(url)
         self.assertEqual(exp, os.path.basename(filename))
 
+    def test_diva_1(self):
+        # Testing redirections from Kungliga biblioteket
+        prov = DiVA(upload=False, verbose=VERBOSE)
+        url = "https://urn.kb.se/resolve?urn=urn:nbn:se:uu:diva-318796"
+        exp = "Lidayova_-_Fast_Methods_for_Vascular_Segmentation_Based_on_Approximate_Skeleton_Detection_2017.pdf"
+        filename = prov.run(url)
+        self.assertEqual(exp, os.path.basename(filename))
+
+    def test_diva_2(self):
+        # Testing absolute URLs and sanitization of filenames
+        prov = DiVA(upload=False, verbose=VERBOSE)
+        url = "https://www.diva-portal.org/smash/record.jsf?pid=diva2%3A1480467"
+        exp = "Alhussein_-_Privacy_by_Design_Internet_of_Things_managing_privacy_2018.pdf"
```

Comment:
When running the test I get `Alhussein_-_Privacy_by_Design_Amp_Internet_of_Things_Managing_Privacy_2018.pdf` which causes it to fail

Reply:
Oh, thanks!